cmd/kubelet/app/server.go

```diff
@@ -1296,3 +1309,28 @@ func parseResourceList(m map[string]string) (v1.ResourceList, error) {
 	}
 	return rl, nil
 }
+
+func configureTracerProvider(s *options.KubeletServer) oteltrace.TracerProvider {
+	if s.KubeletConfiguration.Tracing == nil {
+		return oteltrace.NewNoopTracerProvider()
+	}
+	hostname, err := nodeutil.GetHostname(s.HostnameOverride)
+	if err != nil {
+		klog.V(2).InfoS("Warning: Could not configure tracer provider", "error", err)
+		return oteltrace.NewNoopTracerProvider()
+	}
+	opts := []otlpgrpc.Option{}
+	resourceOpts := []otelsdkresource.Option{
+		otelsdkresource.WithAttributes(
+			semconv.ServiceNameKey.String(hostname),
+			semconv.ServiceInstanceIDKey.String(componentKubelet),
```

Comment:
```suggestion
			semconv.ServiceNameKey.String(componentKubelet),
			semconv.HostNameKey.String(hostname),
```
host name has its own semantic convention, which we can use.  Service name [should be a constant](https://github.com/open-telemetry/opentelemetry-go/blob/8c0951ea94cb20bd9e2338b8fdc20036b8cf995b/semconv/v1.7.0/resource.go#L825).